haskell/haddock.bzl

```diff
@@ -182,6 +258,14 @@ haskell_doc = rule(
       aspects = [haskell_doc_aspect],
       doc = "List of Haskell libraries to generate documentation for.",
     ),
+    "include_transitive_deps": attr.bool(
```

Comment:
This option should actually allow to switch between "include everything" and "include only direct deps" modes. I pushed an update that makes it actually work.

Reply:
Looks reasonable.  I suggest changing the name of this attribute though; at first glance it makes it seem like documentation for transitive deps isn't generated at all (rather than just the index).

Maybe `index_transitive_deps` instead?